fdbclient/ManagementAPI.actor.h

```diff
@@ -248,8 +248,81 @@ bool schemaMatch(json_spirit::mValue const& schema,
 // storage nodes
 ACTOR Future<Void> mgmtSnapCreate(Database cx, Standalone<StringRef> snapCmd, UID snapUID);
 
-Future<Void> addCachedRange(const Database& cx, KeyRangeRef range);
-Future<Void> removeCachedRange(const Database& cx, KeyRangeRef range);
+// Management API written in template code to support both IClientAPI and NativeAPI
+namespace ManagementAPI {
+
+ACTOR template <class DB>
+Future<Void> changeCachedRange(Reference<DB> db, KeyRangeRef range, bool add) {
```

Comment:
Is it possible to put the implementation in ManagementAPI.actor.cpp?

Reply:
template code is usually put in the header files?

Reply:
There are only 2 values of `DB` which share this code, so we can declare both of these in the header file, then put the implementations in `ManagementAPI.actor.cpp`.

Reply:
yeah, I know the way to explicitly instantiate the templates.
Not sure is it adding much benefit here?

Reply:
The compilation time difference seems to be able to ignore. (https://fdb-jenkins.int.snowflakecomputing.com/job/FDB_docker_build/713/flowGraphTable/)
I guess it's okay to just put the implementation here?

Reply:
It's not just the current compile time that is important, but if we want to change the implementation in the future, this should affect as few translation units as possible. I think the current implementation is okay too, though.